docs/open-source/external-action.mdx

```diff
@@ -0,0 +1,250 @@
+- `speak_on_send`: if `True`, then the underlying LLM will generate a message to be spoken into the phone call as the
+  API request is being sent.
+
+- `speak_on_receive`: if `True`, then the Vocode Agent will invoke the underlying
+  LLM to respond based on the result from the API Response or the Error encountered.
+
+- `signature_secret`: a base64 encoded string to enable request validation, see the [Signature Validation section](/open-source/external-action#signature-validation) below for more info
+
+#### `input_schema` Field
+
+The `input_schema` field is a [JSON Schema](https://json-schema.org/) object that instructs how to properly form a payload to send to the External API.
+
+For example, in the [Meeting Assistant Example](/open-source/external-action#meeting-assistant-example) below we formed the following JSON payload:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "length": {
+      "type": "string",
+      "enum": ["30m", "1hr"]
+    },
+    "time": {
+      "type": "string",
+      "pattern": "^d{2}:d0[ap]m$"
+    }
+  }
+}
+```
+
+This is stating the External API is expecting:
+
+- Two fields
+  - `length` (string): either "30m" or "1hr"
+  - `time` (string): a regex pattern defining a time ending in a zero with `am`/`pm` on the end ie: `10:30am`
+
+<Card title="💡 Note" color="#ca8b04">
+    If you’re noticing that this looks very familiar to OpenAI function calling, it is! Vocode treats OpenAI Function Calling as a first-class standard when the agent uses an OpenAI LLM.
+
+    The lone difference is that the top level `input_schema` JSON schema must be an `object` - this is so we can use JSON to send over parameters to the user’s API.
+
+</Card>
+
+#### `description` Field
+
+The `description` is best used to descibe your External Action's purpose. As its passed through directly to the LLM, its the best way to convey instructions to the underlying Vocode Agent.
+
+For example, in the [Meeting Assistant Example](/open-source/external-action#meeting-assistant-example) below we want to book a meeting for 30 minutes to an hour so we set the description as `Book a meeting for a 30 minute or 1 hour call.`
+
+<Card title="💡 Note" color="#ca8b04">
+  The `description` field is passed through and heavily affects how we do our
+  function decisioning so we recommend treating it in the same way you would a
+  prompt to an LLM!
+</Card>
+
+### Responding to External Action API Requests
+
+Once an External Action has been created, the Vocode Agent will issue API requests to the defined `url` during the course of a phone call based on the [configuration noted above](/open-source/external-action#configuring-the-external-action)
+The Vocode API will wait a maximum of _10 seconds_ before timing out the request.
+
+In particular, Vocode will issue a POST request to `url` with a JSON payload that matches `input_schema` , specifically (using the [Meeting Assistant Example](/open-source/external-action#meeting-assistant-example) below):
```

Comment:
"that specifically matches `input_schema`. Using the Meeting Assistant Example below: